ui/src/features/Editor/useCanvasCopyPaste.ts

```diff
@@ -95,21 +83,41 @@ export default ({
 
         if (newNode.type === "batch") {
           parentIdMapArray.push({ prevId: n.id, newId: newNode.id });
-        } else if (newNode.type === "subworkflow") {
+
+          const batchNodeChildren = nodes.filter(
+            (child) => child.parentId === n.id,
+          );
+
+          batchNodeChildren.forEach((child) => {
+            if (!pastedNodes.some((node) => node.id === child.id)) {
```

Comment:
Filtering and then doing forEach separately isn't needed. Just do nodes.forEach and check for parentId === id there.